services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/FileBasedDeviceBackend.java

```diff
@@ -186,7 +194,66 @@ public final void get(final String tenantId, final String type, final String aut
     public void get(final String tenantId, final String type, final String authId, final JsonObject clientContext,
             final Span span,
             final Handler<AsyncResult<CredentialsResult<JsonObject>>> resultHandler) {
-        credentialsService.get(tenantId, type, authId, clientContext, span, resultHandler);
+        credentialsService.get(tenantId, type, authId, clientContext, span, ar -> {
+            if (ar.result().getStatus() == HttpURLConnection.HTTP_NOT_FOUND
+                    && isAutoProvisioningEnabled(type, clientContext)) {
+                provisionDevice(tenantId, authId, clientContext, span, resultHandler);
+            } else {
+                resultHandler.handle(Future.succeededFuture(ar.result()));
```

Comment:
`resultHandler.handle(ar)` should do the trick, shouldn't it?

Reply:
In the current implementation of the `CredentialsService', the AsyncResult will always be successful, but if we assume that this could be changed (as discussed above), then your proposal would change the behavior, wouldn't it? So IMHO we should either leave this as it is *or* add the check if the AsyncResult was successful. WDYT?

Reply:
I do not think that we are changing any behavior here because the `AsyncResult` can already fail FMPOV.